indico/modules/rb/client/js/common/bookings/BookingEditForm.jsx

```diff
@@ -226,10 +237,13 @@ class BookingEditForm extends React.Component {
             />
           )}
           {!room.canUserBook && room.canUserPrebook && isAccepted && (
-            <FormSpy subscription={{dirtyFields: true}}>
-              {({dirtyFields}) =>
+            <FormSpy subscription={{dirtyFields: true, initialValues: true, values: true}}>
+              {({dirtyFields, initialValues, values}) =>
                 Object.keys(dirtyFields).some(
-                  key => key === 'dates' || key === 'timeSlot' || key.startsWith('recurrence.')
+                  key =>
+                    key === 'dates' ||
+                    key.startsWith('recurrence.') ||
+                    this.showWarning(initialValues, values)
```

Comment:
This doesn't really belong here; you are inside the `some()` call, but the check is not specific to any of the dirtyFields key. Move it outside the `some()` call instead.

FWIW, you could change it to `showWarning(dirtyFields, initialValues, values)` and then move the `Object.keys(dirtyFields).some()` check into the function as well.